config/gnu-warnings/noerror-general

```diff
@@ -3,7 +3,6 @@
 # circumstances, so ask the compiler to treat them as errors:
 #
 -Wbad-function-cast
--Wdeclaration-after-statement
```

Comment:
Should this warning also be taken out of error-general file or downgraded to just a warning in that file?
Also, line 2 and 3 of this file should be reworded. Maybe something like:
# HDF5 code should not trigger the following warnings, they are identified
# as candidates for the compiler to treat as errors:

Reply:
There should not be a noerror-general file or any other noerror- file.

The error- and noerror- files are identical under a simple search and replace that is in `config/gnu-flags`: if `PROMOTE_ERRORS` is set to `no` then every occurrence of `-Werror=` is transformed to a warning (`-W`). Looks like the CMake files can do essentially the same thing.

Reply:
"Silence is golden." Declarations after statements are allowed, so let's just delete `-Wdeclaration-after-statement` everywhere it occurs.

Reply:
> There should not be a noerror-general file or any other noerror- file.
> 
> The error- and noerror- files are identical under a simple search and replace that is in `config/gnu-flags`: if `PROMOTE_ERRORS` is set to `no` then every occurrence of `-Werror=` is transformed to a warning (`-W`). Looks like the CMake files can do essentially the same thing.

But wouldn't that promote ALL warnings? I thought the error-general file was intended to only promote those that should pass?

Reply:
> "Silence is golden." Declarations after statements are allowed, so let's just delete `-Wdeclaration-after-statement` everywhere it occurs.

I didn't think we came to a conclusion on this. We would need a way to identify (maybe a ./bin/check_declaration script?) what was valid vs not wanted behavior.

Reply:
I don't know what you mean, "promote ALL warnings." Not all warnings are in the `error-` files. The `error-` files are supposed to contain those warnings that should ordinarily stop the library compilation cold.

Reply:
If there is an objectionable declaration after statement, then the place to flag that is during PR review.

Reply:
> If there is an objectionable declaration after statement, then the place to flag that is during PR review.

I'd rather avoid many "noisy" PRs and just keep the error

Reply:
From a quick review of the platforms being reported in DT, it looks like no configuration is complaining about for loop declarations. This is looking at code that I know have these for loop declarations (blosc filter). Therefore I don't think we need to remove this warning at this time and still allow for loop declarations.
Let's start by implementing for loop declarations and then in a few months take the next step if we all agree?